great_expectations/datasource/data_connector/file_path_data_connector.py

```diff
@@ -92,6 +92,11 @@ def _get_data_reference_list_from_cache_by_data_asset_name(
             )
         )
 
+        if len(self.sorters) > 0:
+            batch_definition_list = self._sort_batch_definition_list(
+                batch_definition_list=batch_definition_list
+            )
+
         path_list: List[str] = [
```

Comment:
Adding sorter logic to `_get_data_reference_list_from_cache_by_data_asset_name()` which will allow the output of sorted batches for `self_check()`